pkg/services/baremetal/baremetal/baremetal.go

```diff
@@ -434,10 +434,16 @@ func (s *Service) chooseHost(ctx context.Context) (
 			continue
 		}
 
-		// This comes last, because now we would choose the machine, but we check
-		// if the config is correct.
+		if host.Spec.RootDeviceHints != nil &&
```

Comment:
can you maybe put this to another PR? Because this is a bugfix that should be visible to users through PR description, commit messages, and release notes. Right now it would not be visible anywhere. 

Additionally, can you add a comment to this that describes the != nil check? Because I actually thought you did the wrong thing here because you don't allow empty rootDeviceHints. You do - but you do it in a hidden way that is not explained. 

However, is this != nil check actually what we want? What about people using a template where rootDeviceHints is actually specified but left empty? Those people get an error here right now without having an idea that they should remove the "rootDeviceHints" key from their host template...